File: exams/segarra/u3/PracticalExam/ec.edu.espe.stateypattern.model/InsertionSort.js
class InsertionSort {
    sort(data) {
        for (let i = 1; i < data.length; i++) {
            let key = data[i];
            let j = i - 1;
            while (j >= 0 && data[j] > key) {
                data[j + 1] = data[j];
                j = j - 1;
            }
            data[j + 1] = key;
        }
        return data;
    }
}

File: exams/arboleda/u3/Strategy/controller/InsertionSort.py
from controller import SortStrategy

# Concrete Strategy: Insertion Sort
class InsertionSort(SortStrategy):
    def sort(self, data):
        for i in range(1, len(data)):
            key = data[i]
            j = i - 1
            while j >= 0 and key < data[j]:
                data[j + 1] = data[j]
                j -= 1
            data[j + 1] = key
        return data